DA-141: Correct terminology in front- and backend #2, refactor scheme…

JeannedArk opened the pull request:
…UploadModal

The terminology was further corrected, LinkSet was renamed to LinkType. The
schemeUploadModal and schemeViewModal was refactored. Instead of converting
between two different models (front- and backend) only the backend model
is used now, so no conversion is needed anymore, which caused an overhead
of code. Furthermore they were generally refactored, including creating a
UtilityService. The scheme upload with JSON files was also fixed.

JeannedArk commented:
@judembo This is the commit which is supposed to fix the upload.

judembo commented:
Everything seems to work fine now and terminology appears to be consistent.
Only exporting a project to XMI still uses the old terminology (sorry if I'm being picky)

judembo commented on src/main/webapp/controllers/schemes/schemeUploadModalController.js:
```diff
@@ -25,16 +25,16 @@ angular
             $rootScope.alertsModal = [];
+            $scope.visKind = 'None';
+            $scope.positioning = undefined;
             $scope.targetIdCounter = 0;
```
is targetIdCounter still needed?

JeannedArk replied:
No, you are right and therefore removed.

JeannedArk committed:
DA-141: Correct terminology in front- and backend #2, refactor schemeUploadModal

The terminology was further corrected, LinkSet was renamed to LinkType. The
schemeUploadModal and schemeViewModal was refactored. Instead of converting
between two different models (front- and backend) only the backend model
is used now, so no conversion is needed anymore, which caused an overhead
of code. Furthermore they were generally refactored, including creating a
UtilityService. The scheme upload with JSON files was also fixed.

JeannedArk commented:
No, you are not picky! 👍 I will merge it so everybody can continue working, because of the great amount of changes again. We never checked the XML export and import again, but we can do this with this ticket https://github.com/annefried/discanno/issues/3
